test_spack.py

```diff
@@ -43,6 +43,25 @@ def Run(self, command: str, cwd='.', parallel=False):
     def Srun(self, command: str, cwd='.'):
         return self.Run(command, cwd, parallel=True)
 
+    def spack_install_and_test(self, command: str, cwd='.'):
+        if 'cosmo@dev-build' in command:
+            spack_install = 'spack devbuildcosmo'
+        elif 'cosmo' in command and 'cosmo-dycore' not in command:
+            spack_install = 'spack installcosmo'
+        else:
+            spack_install = 'spack install'
+
+        if '--show-log-on-error' in command:
```

Comment:
Nice catch! De facto we don't show the log on error for all of your tests.
But I think we actually should show them for all tests. Do you agee?
Which means add `--show-log-on-error` to `command` in here and remove it from the test code further down.